### PR TITLE
Add `MakeDirectory` Predicate

### PR DESCRIPTION
**Pull request by gmarceau**

### CODE REVIEW

File: plumbum/cli/switches.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 from plumbum.lib import six, getdoc
 from plumbum import local
+from plumbum.cmd import mkdir
```
Comment (henryiii) [inline plumbum/cli/switches.py]: path's already have a mkdir.

File: plumbum/cli/switches.py [hunk 2]
```diff
@@ -416,6 +417,15 @@ def ExistingDirectory(val):
         raise ValueError("%r is not a directory" % (val,))
     return p
 
+
+@Predicate
+def MakeDirectory(val):
+    p = local.path(val)
+    if p.is_file():
+        raise ValueError('%r is a file, should be nonexistent, or a directory' % (val,))
+    mkdir('-p', p)
```
Comment (henryiii) [inline plumbum/cli/switches.py]: `p.mkdir()` should work here.

Comment (henryiii) [reply plumbum/cli/switches.py]: That is embarrassing; I was going to ask you to add a test for this, and was going to show you an example, but apparently none of the predicates are tested. 😕 So I'll take this without tests and I'll add them for all four later (unless you'd like to!)
